mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir

```diff
@@ -4115,3 +4115,39 @@ func.func @step_scalable() -> vector<[4]xindex> {
   %0 = vector.step : vector<[4]xindex>
   return %0 : vector<[4]xindex>
 }
+
+// -----
+
+// CHECK-LABEL: @extract_arith_constnt
+func.func @extract_arith_constnt() -> i32 {
+  %v = arith.constant dense<0> : vector<32x1xi32>
+  %c_0 = arith.constant 0 : index
+  %elem = vector.extract %v[%c_0, %c_0] : i32 from vector<32x1xi32>
+  return %elem : i32
+}
+
+// CHECK: %[[VAL_0:.*]] = arith.constant dense<0> : vector<32x1xi32>
+// CHECK: %[[VAL_1:.*]] = builtin.unrealized_conversion_cast %[[VAL_0]] : vector<32x1xi32> to !llvm.array<32 x vector<1xi32>>
+// CHECK: %[[VAL_2:.*]] = llvm.extractvalue %[[VAL_1]][0] : !llvm.array<32 x vector<1xi32>>
+// CHECK: %[[VAL_3:.*]] = llvm.mlir.constant(0 : i64) : i64
+// CHECK: %{{.*}} = llvm.extractelement %[[VAL_2]]{{\[}}%[[VAL_3]] : i64] : vector<1xi32>
+
+// -----
+
+// CHECK-LABEL: @insert_arith_constnt()
+
+func.func @insert_arith_constnt() -> vector<32x1xi32> {
+  %v = arith.constant dense<0> : vector<32x1xi32>
+  %c_0 = arith.constant 0 : index
+  %c_1 = arith.constant 1 : i32
+  %v_1 = vector.insert %c_1, %v[%c_0, %c_0] : i32 into vector<32x1xi32>
+  return %v_1 : vector<32x1xi32>
+}
+    
+// CHECK: %[[VAL_0:.*]] = arith.constant dense<0> : vector<32x1xi32>
+// CHECK: %[[VAL_1:.*]] = builtin.unrealized_conversion_cast %[[VAL_0]] : vector<32x1xi32> to !llvm.array<32 x vector<1xi32>>
+// CHECK: %[[VAL_2:.*]] = arith.constant 1 : i32
+// CHECK: %[[VAL_3:.*]] = llvm.extractvalue %[[VAL_1]][0] : !llvm.array<32 x vector<1xi32>>
+// CHECK: %[[VAL_4:.*]] = llvm.mlir.constant(0 : i64) : i64
+// CHECK: %[[VAL_5:.*]] = llvm.insertelement %[[VAL_2]], %[[VAL_3]]{{\[}}%[[VAL_4]] : i64] : vector<1xi32>
+// CHECK: %{{.*}} = llvm.insertvalue %[[VAL_5]], %[[VAL_1]][0] : !llvm.array<32 x vector<1xi32>>
```

Comment:
I'm not sure if a patch for folding should be adding new tests in vector-to-llvm.mlir. Can you instead add these to mlir/test/Dialect/Vector/canonicalize.mlir ?

Reply:
https://github.com/llvm/llvm-project/blob/988978f964fb84cb99c83e6cd260dcc395afb6c2/mlir/test/Dialect/Vector/canonicalize.mlir#L469

As you can see from the test here, `canonicalize` will just eliminate `insertOp` and `extractOp`, so it seems pointless to add the test here. But I think it makes sense to include the test in `vector-to-llvm.mlir`, which itself is where I found the problem in running this pass, in addition to the other tests where this fold's functionality is demonstrated. 
It should determine if it can be folded after each run of the pattern.

Reply:
Both kinds of tests are needed here:
1. We must have a "unit" test in `canonicalize.mlir` (or maybe there's a way to test only folding) that exercises exclusively these folders. I suppose you need a separate test for insert and another for extract so they don't cancel out.
2. We want a "regression" test in the pass to ensure the originally observed incorrect behavior is fixed and remains so.

Reply:
Testing insert or extract alone may not work, as insert or extract will be eliminated directly, just like the test I wrote.